mesonbuild/environment.py

```diff
@@ -126,6 +130,10 @@ def compute_llvm_sufix(coredata: coredata.CoreData):
         if suffix != exe:
             return suffix
 
+    # If either C or C++ compiler is Clang and we get here, they're unsuffixed. Force this.
+    if ('cpp' in compilers and compilers['cpp'].id == 'clang') or ('c' in compilers and compilers['c'].id == 'clang'):
```

Comment:
Instead of recomputing these, just save them off.

```python
cpp_compiler_is_clang = 'cpp' in compilers and compilers['cpp'].id == 'clang'
c_compiler_is_clang = 'c' in compilers and compilers['c'].id == 'clang'
```